src/React.re

```diff
@@ -886,6 +887,73 @@ external useDebugValue: ('value, ~format: 'value => string=?, unit) => unit =
 
 module Experimental = {
   /* This module is used to bind to APIs for future versions of React. There is no guarantee of backwards compatibility or stability. */
+  /* https://react.dev/reference/react/use */
+  [@mel.module "react"] external usePromise: Js.Promise.t('a) => 'a = "use";
+  [@mel.module "react"] external useContext: Context.t('a) => 'a = "use";
+  /* https://react.dev/reference/react/useTransition */
+  [@mel.module "react"]
+  external useTransitionAsync:
+    unit => (bool, callbackAsync(callbackAsync(unit, unit), unit)) =
+    "useTransition";
+
+  /* https://react.dev/reference/react/useOptimistic */
+  [@mel.module "react"]
+  external useOptimistic:
+    ('state, ('state, 'optimisticValue) => 'state) =>
+    ('state, 'optimisticValue => unit) =
+    "useOptimistic";
+
+  module FormData = {
+    /* This file is embeded since https://github.com/melange-re/melange/pull/1153 gets merged */
+
+    type t;
+    type file;
+    type blob;
+    type entryValue;
+
+    [@mel.new] external make: unit => t = "FormData";
+    [@mel.send.pipe: t] external append: (string, string) => unit = "append";
+    [@mel.send.pipe: t] external delete: string => unit = "delete";
+    [@mel.send.pipe: t] external get: string => option(entryValue) = "get";
+    [@mel.send.pipe: t]
+    external getAll: string => array(entryValue) = "getAll";
+    [@mel.send.pipe: t] external set: (string, string) => unit = "set";
+    [@mel.send.pipe: t] external has: string => bool = "has";
+    [@mel.send] external keys: t => Js.Iterator.t(string) = "keys";
+    [@mel.send] external values: t => Js.Iterator.t(entryValue) = "values";
+
+    [@mel.send.pipe: t]
+    external appendObject: (string, Js.t({..}), ~filename: string=?) => unit =
+      "append";
+
+    [@mel.send.pipe: t]
+    external appendBlob: (string, blob, ~filename: string=?) => unit =
+      "append";
+
+    [@mel.send.pipe: t]
+    external appendFile: (string, file, ~filename: string=?) => unit =
+      "append";
+
+    [@mel.send.pipe: t]
+    external setObject: (string, Js.t({..}), ~filename: string=?) => unit =
+      "set";
+
+    [@mel.send.pipe: t]
+    external setBlob: (string, blob, ~filename: string=?) => unit = "set";
+
+    [@mel.send.pipe: t]
+    external setFile: (string, file, ~filename: string=?) => unit = "set";
+
+    [@mel.send]
+    external entries: t => Js.Iterator.t((string, entryValue)) = "entries";
+  };
 
-  [@mel.module "react"] external use: Js.Promise.t('a) => 'a = "use";
+  type formAction;
+
+  /* https://react.dev/reference/react/useActionState */
+  [@mel.module "react"]
+  external useActionState:
+    (('state, FormData.t) => unit, 'state, ~permalink: bool=?) =>
```

Comment:
should we add a unit argument at the end?

Reply:
Since I don't want to push code to this branch directly, done here: https://github.com/reasonml/reason-react/pull/867